bentoml/_internal/server/service_app.py

```diff
@@ -244,24 +251,30 @@ def middlewares(self) -> t.List["Middleware"]:
 
         return middlewares
 
+    @property
+    def on_startup(self) -> t.List[t.Callable[[], None]]:
+        on_startup = super().on_startup
+        on_startup.insert(0, self.bento_service._on_asgi_app_startup)
+        return on_startup
+
+    @property
+    def on_shutdown(self) -> t.List[t.Callable[[], None]]:
+        on_shutdown = super().on_shutdown
+        on_shutdown.insert(0, self.bento_service._on_asgi_app_shutdown)
+        return on_shutdown
+
     def __call__(self) -> "Starlette":
-        from starlette.applications import Starlette
-
-        app = Starlette(
-            debug=True,  # TDOO: inject this from `debug=True` or `--production` flag
-            routes=self.routes(),
-            middleware=self.middlewares(),
-            on_shutdown=[self.bento_service._on_asgi_app_shutdown],
-            on_startup=[self.bento_service._on_asgi_app_startup, self.mark_as_ready],
-        )
+        app = super().__call__()
 
         for mount_app, path, name in self.bento_service._mount_apps:
-            app.mount(app=mount_app, path=path, name=name)
+            app.mount(
+                app=mount_app, path=path, name=name
+            )  # TODO(jiang): mount wsgi app ?
```

Comment:
wsgi app is already supported, there's no change here needed, https://github.com/bentoml/BentoML/blob/bentoml-1.0/bentoml/_internal/service/service.py#L190

Reply:
This is more about typing

Reply:
```suggestion
            )
```